src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -113,6 +113,9 @@ class Crystal::Doc::Generator
     return false if nodoc?(type)
     return true if crystal_builtin?(type)
 
+    # Don't include lib types or types inside a lib type
+    return false if type.is_a?(Crystal::LibType) || type.namespace.is_a?(LibType)
```

Comment:
Can there be multiple levels of nesting inside lib types? Does this handle it?

Reply:
Nope, there's only one level of nesting. I thought about that too when writing that line :-)